controller/engine_image_controller.go

```diff
@@ -209,7 +209,8 @@ func (ic *EngineImageController) syncEngineImage(key string) (err error) {
 		return err
 	}
 
-	if engineImage.Spec.OwnerID == "" {
+	if nodeStatusDown, _ := ic.ds.IsNodeDownOrDeleted(engineImage.Spec.OwnerID); nodeStatusDown == true ||
```

Comment:
If we know the `OwnerID` is empty, we won't want to check if the node is down or not.

We can use a more explicit way, only check node if the owner is not empty.